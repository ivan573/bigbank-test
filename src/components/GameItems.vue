<template>
  <ul class="game-items">
    <li v-for="item in items" class="game-items__item" :key="item.id" @click="handleItemClick(item.id)">
      <p class="game-items__item-data">{{ item.name }}</p>
      <p class="game-items__item-data">Price: {{ item.cost }}</p>
    </li>
  </ul>
</template>

<script>
export default {
  name: 'GameItems',
  props: {
    items: Array
  },
  methods: {
    handleItemClick(id) {
      this.$emit('itemClick', id)
    }
  }
}
</script>

<style lang="scss" scoped>
  .game-items {
    padding: 0;
    display: flex;
    flex-wrap: wrap;
    gap: 8px;
    list-style: none;

    &__item {
      padding: 8px;
      border: $border;
      border-radius: 8px;
      cursor: pointer;
      background-color: white;
      transition: 0.1s;

      &:hover {
        transform: scale(1.2);
      }
    }
  }

</style>
