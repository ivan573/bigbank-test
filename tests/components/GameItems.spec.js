import { shallowMount } from '@vue/test-utils'
import GameItems from '@/components/GameItems.vue'

describe('GameItems.vue', () => {
  const items = [
    {
      cost: 50,
      id: 'hpot',
      name: 'Healing potion'
    },
    {
      cost: 100,
      id: 'tricks',
      name: 'Book of Tricks'
    },
    {
      cost: 300,
      id: 'mtrix',
      name: 'Book of Megatricks'
    }
  ]

  const wrapper = shallowMount(GameItems, {
    propsData: { items }
  })

  const itemElements = wrapper.findAll('.game-items__item')

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('displays all of the items', () => {
    expect(itemElements.length).toBe(items.length)
  })

  it('displays all item data correctly', () => {
    itemElements.wrappers.forEach((element, index) => {
      const itemDataElements = element.findAll('.game-items__item-data').wrappers
      expect(itemDataElements[0].text()).toBe(items[index].name)
      expect(itemDataElements[1].text()).toBe(`Price: ${items[index].cost}`)
    })
  })

  it('emits itemClick event with the item id on item click', () => {
    itemElements.wrappers.forEach((element, index) => {
      element.trigger('click')
      expect(wrapper.emitted().itemClick[index]).toEqual([items[index].id])
    })
  })
})
