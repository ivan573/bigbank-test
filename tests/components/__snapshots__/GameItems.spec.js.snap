// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameItems.vue matches snapshot 1`] = `
<ul
  class="game-items"
>
  <li
    class="game-items__item"
  >
    <p
      class="game-items__item-data"
    >
      Healing potion
    </p>
     
    <p
      class="game-items__item-data"
    >
      Price: 50
    </p>
  </li>
  <li
    class="game-items__item"
  >
    <p
      class="game-items__item-data"
    >
      Book of Tricks
    </p>
     
    <p
      class="game-items__item-data"
    >
      Price: 100
    </p>
  </li>
  <li
    class="game-items__item"
  >
    <p
      class="game-items__item-data"
    >
      Book of Megatricks
    </p>
     
    <p
      class="game-items__item-data"
    >
      Price: 300
    </p>
  </li>
</ul>
`;
